### DISCUSSION TITLE
Implicit zero copy on `Vec<u8>`

### DISCUSSION THREAD

**Discussion by temeddix** [fzyzcjy/flutter_rust_bridge#1222] [Ideas]
After #1221, I wondered why we would need to copy `Vec<u8>` when we have the ability to 'zero copy' the bytes array. What if we could just zero copy `Vec<u8>` types by default without explicit `ZeroCopyBuffer`? If I can contribute, would it be okay to work on this idea and deprecate `ZeroCopyBuffer`?

If this is accepted, proper documentation and depreciation warning should be added.

I know that the idea of 'binding unmodified rust' is ongoing, but maybe this change could be applied prior to that.

**Comment by fzyzcjy** [comment]
Good question! IIRC, I did that at the beginning is probably because of simplicity of the code generator - the ZeroCopyBuffer is in allo_isolate. I only need to write naive code generator that passes this around.

If we are going to change the default behavior, I guess we can contribute to https://github.com/sunshine-protocol/allo-isolate's ZeroCopyBuffer-related things.
